mllib/src/main/scala/org/apache/spark/ml/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -115,7 +122,14 @@ private[spark] object DecisionTreeMetadata extends Logging {
     }
     require(numFeatures > 0, s"DecisionTree requires number of features > 0, " +
       s"but was given an empty features vector")
-    val numExamples = input.count()
+    val (numExamples, weightSum) = input.aggregate((0L, 0.0))(
+      (acc, x) => (acc, x) match {
```

Comment:
I think the match statement is redundant? just `((count, weight), instance) => ...`? I forget whether there's some scala reason that won't work but you're just providing a function with this arg anyway

Reply:
hmm, that didn't seem to work, but I've updated in a slightly different way, hope it's clearer